File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int>> nums2;
        int i=0,j=nums.size()-1;
        for(i;i<nums.size();i++){
            nums2.push_back(make_pair(nums[i],i));
        }
        i=0;
        sort(nums2.begin(), nums2.end());
        while(i<j){
            if(nums2[i].first+nums2[j].first == target){
                return {min(nums2[i].second, nums2[j].second),max(nums2[i].second,nums2[j].second)};
            }
            else if(nums2[i].first+nums2[j].first > target){
                j--;
            }
            else{
                i++;
            }
        }
        return {};
    }
};

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
        int left = 0, right = nums.length-1, mid;
        while(left < right){
            mid = (left + right)/2;
            if(nums[mid] == target) return mid;
            if(nums[mid] > target){
                if(nums[left] > target && nums[mid] > nums[right]){
                    left = mid + 1;
                } else{
                    right = mid;
                }
            }
            else{
                if(nums[right] >= target || nums[mid] > nums[right]){
                    left = mid + 1;
                } else {
                    right = mid;
                }
            }
        }
        return nums[left] == target? left : -1;
    }
}
